lib/Doctrine/ORM/Configuration.php

```diff
@@ -748,6 +749,60 @@ public function setCustomDatetimeFunctions(array $functions)
         }
     }
 
+    /**
+     * Sets a map of php typed fields to DBAL types for auto-completion.
+     *
+     * Keys must be FQCNs of the PHP types
+     * Values are either FQCNs of DBAL Type or their name
+     *
+     * Any previously added mappings are discarded.
+     *
+     * @param array $mappings The map of custom DQL date/time functions.
+     * @psalm-param array<class-string, class-string<Type>|string> $mappings
+     *
+     * @return void
+     */
+    public function setTypedFieldMappings(array $mappings)
+    {
+        foreach ($mappings as $phpType => $dbalType) {
+            $this->addTypedFieldMapping($phpType, $dbalType);
+        }
+    }
+
+    /**
+     * @psalm-param class-string              $phpType  the FQCN of PHP type to be mapped
+     * @psalm-param class-string<Type>|string $dbalType either FQCN of DBAL Type or its name
+     *
+     * @return void
+     */
+    public function addTypedFieldMapping(string $phpType, string $dbalType)
+    {
+        $this->_attributes['typedFieldMappings'][$phpType] = $dbalType;
+    }
+
+    /**
+     * Gets the mapped DBAL Type for given PHP type
+     *
+     * @psalm-param class-string $phpType  the FQCN of PHP type to be mapped
+     *
+     * @return string|null
+     * @psalm-return class-string<Type>|string|null either FQCN of DBAL Type or its name
+     */
+    public function getTypedFieldMapping($phpType)
+    {
+        return $this->_attributes['typedFieldMappings'][$phpType] ?? null;
+    }
```

Comment:
This method is never used.

Reply:
it is not at the moment, but I think its useful method and it follows the pattern of other other functions like `set/get/addCustomHydrationMode(s)`  or `set/get/addCustomDatetimeFunction(s)`. Id like to keep it.